server/lomas_server/dp_queries/dp_querier.py

```diff
@@ -22,27 +28,30 @@ def __init__(self, data_connector: DataConnector) -> None:
         self.data_connector = data_connector
 
     @abstractmethod
-    def cost(self, query_json: BaseModel) -> tuple[float, float]:
+    def cost(self, query_json: RequestModelGeneric) -> tuple[float, float]:
         """
         Estimate cost of query.
 
         Args:
-            query_json (BaseModel): The JSON request object for the query.
-
+            query_json (RequestModelGeneric): The input object of the request.
+                Must be a subclass of RequestModel.
         Returns:
             tuple[float, float]: The tuple of costs, the first value is
                 the epsilon cost, the second value is the delta value.
         """
 
     @abstractmethod
-    def query(self, query_json: BaseModel) -> str:
+    def query(
+        self, query_json: QueryModelGeneric
+    ) -> dict | int | float | List[Any] | Any | str:
```

Comment:
maybe (in the future) we should uniformise this for all libs

Reply:
Yes I agree. I was also wondering if we should do "output" models for replies to the client. If we define these in a core library, the client could simply do a model_validate when receiving a response.